modules/oe_translation_corporate_workflow/modules/oe_translation_active_revision/tests/src/FunctionalJavascript/ActiveRevisionTest.php

```diff
@@ -0,0 +1,573 @@
+    $default_values = [
+      'major' => 0,
+      'minor' => 1,
+      'patch' => 0,
+    ];
+    \Drupal::service('entity_version.entity_version_installer')->install('node', ['page'], $default_values);
+
+    \Drupal::entityTypeManager()->getStorage('entity_version_settings')->create([
+      'target_entity_type_id' => 'node',
+      'target_bundle' => 'page',
+      'target_field' => 'version',
+    ])->save();
+
+    \Drupal::service('router.builder')->rebuild();
+
+    $user = $this->setUpTranslatorUser();
+    // Grant the editorial roles.
+    foreach (['oe_author', 'oe_reviewer', 'oe_validator'] as $role) {
+      $user->addRole($role);
+      $user->save();
+    }
+    $this->drupalLogin($user);
+  }
+
+  /**
+   * Tests that when we validated/publish, we can map the translations.
+   */
+  public function testActiveRevisionMapping(): void {
+    /** @var \Drupal\node\NodeStorageInterface $node_storage */
+    $node_storage = $this->entityTypeManager->getStorage('node');
+
+    // Create a node, validate it, and translate it.
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $node_storage->create([
+      'type' => 'page',
+      'title' => 'My node',
+      'field_non_translatable_field' => 'Non translatable value',
+      'moderation_state' => 'draft',
+    ]);
+    $node->save();
+
+    $node = $node_storage->load($node->id());
+    $node = $this->moderateNode($node, 'validated');
+
+    $node->addTranslation('fr', ['title' => 'My FR node']);
+    $node->save();
+
+    // Go to the node moderation state form and assert we don't see our
+    // elements for mapping the active revision.
+    $this->drupalGet($node->toUrl());
+    $this->assertSession()->fieldNotExists('The new version needs NEW translations');
+    $this->assertSession()->fieldNotExists('The new version does NOT need NEW translations (there have been only changes to non-translatable fields)');
+
+    $node->delete();
+
+    // Now create a node for each target moderation state and assert we will
+    // get our form elements to choose what to do with the translations upon
+    // making a new revision.
+    $states = [
+      'Validated',
+      'Published',
+    ];
+
+    foreach ($states as $state) {
+      // Create a node and translate it.
+      /** @var \Drupal\node\NodeInterface $node */
+      $node = $node_storage->create([
+        'type' => 'page',
+        'title' => 'My node',
+        'field_non_translatable_field' => 'Non translatable value',
+        'moderation_state' => 'draft',
+      ]);
+      $node->save();
+
+      $node = $node_storage->load($node->id());
+      $node = $this->moderateNode($node, 'published');
+
+      $node->addTranslation('fr', ['title' => 'My FR node']);
+      $node->save();
+
+      // Keep track of the version 1 revision ID.
+      $version_one_revision_id = $node->getRevisionId();
+
+      $this->drupalGet($node->toUrl());
+      $this->assertSession()->pageTextContains('My node');
```

Comment:
Here we may again assert that we still don't see form elements for mapping the active revision.